core/pkg/ingress/controller/controller.go

```diff
@@ -509,9 +509,13 @@ func (ic *GenericController) getBackendServers(ingresses []*extensions.Ingress)
 				ca := ic.annotations.CertificateAuth(ing)
 				if ca != nil {
 					server.CertificateAuth = *ca
```

Comment:
When building the annotation we just check if the secret exists in GetAuthCertificate, then cast it into an *ingress.SSLCert and then build the AuthSSLCert, but there are paths if the secret is created with `tls.crt` and `tls.key` but no `ca.crt` that CAFileName will be "".